mesheryctl/internal/cli/root/system/component.go

```diff
@@ -0,0 +1,489 @@
+var searchComponentsCmd = &cobra.Command{
+	Use:   "search",
+	Short: "search registered components",
+	Long:  "search components registered in Meshery Server",
+	Example: `
+// Search for components using a query
+mesheryctl exp components search [query-text]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	Args: func(_ *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp component search [query-text]\nRun 'mesheryctl exp component search --help' to see detailed help message"
+		if len(args) == 0 {
+			return fmt.Errorf("search term is missing\n\n%v", errMsg)
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			log.Fatalln(err, "error processing config")
+		}
+
+		baseUrl := mctlCfg.GetBaseMesheryURL()
+		queryText := args[0]
+
+		url := fmt.Sprintf("%s/api/meshmodels/components?search=%s&pagesize=all", baseUrl, queryText)
+		req, err := utils.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		resp, err := utils.MakeRequest(req)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		// defers the closing of the response body after its use, ensuring that the resources are properly released.
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		componentsResponse := &models.MeshmodelComponentsAPIResponse{}
+		err = json.Unmarshal(data, componentsResponse)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		header := []string{"Model", "kind", "Version"}
+		rows := [][]string{}
+
+		for _, component := range componentsResponse.Components {
+			if len(component.DisplayName) > 0 {
+				rows = append(rows, []string{component.Model.Name, component.Kind, component.APIVersion})
+			}
+		}
+
+		if len(rows) == 0 {
+			// if no component is found
+			fmt.Println("No components(s) found")
+			return nil
+		} else {
+			utils.PrintToTable(header, rows)
+		}
+
+		return nil
+	},
+}
+
+// ComponentsCmd represents the mesheryctl exp components command
+var ComponentsCmd = &cobra.Command{
+	Use:   "components",
+	Short: "View list of components and detail of components",
+	Long:  "View list of components and detailed information of a specific component",
+	Example: `
+// To view list of components
+mesheryctl exp components list
+// To view a specific component
+mesheryctl exp components view [component-name]
```

Comment:
How is this different from 
https://github.com/meshery/meshery/pull/10118/files#diff-09fcaa361285a17c501b91b394c9c6ef119700f161e9ac10f7eb6572bde16336R190
?

Reply:
This is the entry point for the components command. Here we are showing the example of how to use `mesheryctl exp component` command.